### DISCUSSION TITLE
On Authorize User allow enforcing a single store + avoid entering store id

### DISCUSSION THREAD

**Discussion by ndeet** [btcpayserver/btcpayserver#3121] [Technical Support]
Currently on the [Authorize User endpoint](https://docs.btcpayserver.org/API/Greenfield/v1/#tag/Authorization) we can set certain options on the url generation. One important piece missing here at least for the use cases of online stores is that we should be able to enforce a single store. Currently there is only `selectiveStores` but this will allow the user to select one or more stores, not setting `selectiveStores` will generate an api-key for all stores which is not ideal from a security perspective.

An additional benefit if we allow enforcing a single store the response payload from BTCPay to the redirect url could contain that store ID. This would be a huge UX improvement as the user does not need to go to BTCPay again to find out the store ID. 

Current flow:
- user enters BTCPay url in shop
- gets redirected to BTCPay api-keys/authorize
- selects store(s) 
- gets redirected back to shop
- needs to enter the store ID, user needs to open up BTCPay in second tab to check
- user enters store ID per copy&paste
- done

New flow with e.g. `enforceSingleStore` option:
- user enters BTCPay url in shop
- gets redirected to BTCPay api-keys/authorize
- can only select a single store (if user has only one store it is preselected)
- gets redirected back to shop
- response data contains store ID we can use that and no further user interaction needed
- done

**Comment by BitcoinMitchell** [comment]
In the new rewrite for Prestashop, I've added validation to ensure a single store is selected and all the permissions are for this one store, but it would be much nicer if BTCPay Server just told me the `StoreID` of the _one_ store that the user selected.

**Comment by bolatovumar** [comment]
@ndeet Is the idea that even if the user requests multiple permissions like `btcpay.store.webhooks.canmodifywebhooks` and `btcpay.store.canmodifystoresettings` they would still only be able to select one store for all of them? Currently the user can select any number of stores for any number of permissions but what you are suggesting is that we allow user to select only one store total for N permissions?

**Reply by BitcoinMitchell** [reply]
@bolatovumar That is exactly what I would like (and I'm pretty sure that is what @ndeet wants as well).

**Reply by bolatovumar** [reply]
One thing to consider here is that currently you can select stores only if a permission is a store for a store policy. For each store policy you can pick any number of stores to authorize (e.g., for `btcpay.store.webhooks.canmodifywebhooks` you can authorize Store 1 and 2 but for `btcpay.store.canmodifystoresettings` only Store 1).

If we were to make the changes suggested here, would we be grouping all store-related permissions into one which would have only one store select?

See current experience here (notice the third option doesn't allow you to pick a store because it is not a store policy):

![example](https://user-images.githubusercontent.com/1934678/143661494-757bc73b-6184-4957-b328-393d2f8d9175.PNG)

**Reply by ndeet** [reply]
Thanks for your thoughts @bolatovumar Exactly, ideally you only have one store to select at the top (instead to select store(s) for each permission) which may would lead to what @Kukks suggested below - to have a separate flow/screen for that use case? We can also leave the current UI but then we need to limit and make sure only same one store is selected for each permission. Also we only need store specific permissions here. The others are global anyway.

**Comment by Kukks** [comment]
I would build the suggested flow as a new option or a separate flow

**Comment by Kukks** [comment]
This is probably a duplicate of #2902 btw
